TESTS/mbed_drivers/timer/main.cpp

```diff
@@ -30,35 +30,24 @@ using namespace utest::v1;
 
 extern uint32_t SystemCoreClock;
 
-/* Macro to define delta based on CPU clock frequency.
- *
- * Note that some extra time is counted by the timer.
- * Additional time is caused by the function calls and
- * additional operations performed by wait and
- * stop functions before in fact timer is stopped. This may
- * add additional time to the counted result.
- *
- * To take in to account this extra time we introduce DELTA
- * value based on CPU clock (speed):
- *   DELTA = TOLERANCE_FACTOR / SystemCoreClock * US_FACTOR
- *
- *   e.g.
- *   For K64F          DELTA = (30000 / 120000000) * 1000000 = 250[us]
- *   For NUCLEO_F070RB DELTA = (30000 /  48000000) * 1000000 = 625[us]
- *   For NRF51_DK      DELTA = (30000 /  16000000) * 1000000 = 1875[us]
- */
 #define US_PER_SEC       1000000
 #define US_PER_MSEC      1000
-#define TOLERANCE_FACTOR 30000.0f
-#define US_FACTOR        1000000.0f
+#define MSEC_PER_SEC     1000
 
-static const int delta_sys_clk_us = ((int) (TOLERANCE_FACTOR / (float)SystemCoreClock * US_FACTOR));
-
-/* When test performs time measurement using Timer in sequence, then measurement error accumulates
- * in the successive attempts. */
- #define DELTA_US(i) (delta_sys_clk_us * i)
- #define DELTA_S(i)  ((float)delta_sys_clk_us * i / US_PER_SEC)
- #define DELTA_MS(i) (1 + ( (i * delta_sys_clk_us) / US_PER_MSEC))
+ /*
+ * Define tolerance as follows:
+ * tolerance = 500 us + 2% of measured time
+ *
+ * e.g.
+ * 1 ms delay: tolerance = 520 us
+ * 10 ms delay: tolerance = 700 us
+ * 100 ms delay: tolerance = 2500 us
+ * 1000 ms delay: tolerance = 20500 us
+ *
```

Comment:
This new way of defining tolerance seems like a pretty big change from the previous model, will have to check with the relevant people in the team